book/invalidation.md

```diff
@@ -290,31 +308,33 @@ This makes the `BlockLayout`'s `layout` function idempotent because
 each call will assign a new `children` array.
 
 Let's check all other `layout` methods for idempotency by reading them
-and noting any non-idempotent method calls. I found:[^exercises]
+and noting any subroutine calls. I found:[^exercises]
 
 [^exercises]: If you've being doing exercises throughout this book,
     there might be more, in which case it might be harder to follow
     along, depending on how you've implemented them. But the core idea
     is replacing non-idempotent calls with idempotent ones.
 
-- In `new_line`, `BlockLayout` will append to its `children` array;
+- In `new_line`, `BlockLayout` will append to its `children` array.
 - In `text` and `input`, `BlockLayout` will append to the `children`
-  array of some `LineLayout` child
+  array of some `LineLayout` child.
 - In `text` and `input`, `BlockLayout` will call `get_font`, as will
-  the `TextLayout` and `InputLayout` methods
-- Basically every layout method calls `display_px`
+  the `TextLayout` and `InputLayout` methods.
+- Basically every layout method calls `device_px`.
 
 Luckily, the `new_line` and `add_inline_child` methods are only called
 through `layout`, which resets the `children` array. Meanwhile,
-`get_font` acts as a cache, so multiple calls return the same font
+`get_font` acts as a cache,^[The presence of a cache is a sure sign that
+idempotent methods are present.] so multiple calls return the same font
```

Comment:
I'm not sure what this comment means... I guess that the stuff being protected (the font subsystem) is already idempotent?

Reply:
Removed then